test/extensions/filters/listener/proxy_protocol/proxy_protocol_test.cc

```diff
@@ -210,6 +212,23 @@ INSTANTIATE_TEST_SUITE_P(IpVersions, ProxyProtocolTest,
                          testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
                          TestUtility::ipTestParamsToString);
 
+// This test ensures the socket file event was reset after timeout, otherwise
+// the assertion which avoid to create file event duplicated will be triggered.
+TEST_P(ProxyProtocolTest, Timeout) {
+  connect();
+  absl::SleepFor(absl::Seconds(5));
+  dispatcher_->run(Event::Dispatcher::RunType::NonBlock);
+  if (GetParam() == Envoy::Network::Address::IpVersion::v4) {
+    EXPECT_EQ(server_connection_->addressProvider().remoteAddress()->ip()->addressAsString(),
+              "127.0.0.1");
+  } else {
+    EXPECT_EQ(server_connection_->addressProvider().remoteAddress()->ip()->addressAsString(),
+              "::1");
+  }
+
+  disconnect();
+}
```

Comment:
I'm not seeing any expectations in the test that verify that the timeout was hit and the filter chain was then re-created.  Are there some stats that can be checked in order to verify that this test does as advertised?

Reply:
yea, let me see how to verfiy that.